lib/wadler/print.rb

```diff
@@ -160,13 +204,24 @@ def group(indent = 0, open_obj = '', close_obj = '',
       tokens << Oppen.end
     end
 
-    # Create a new nest.
+    # Creating a new non-grouping indentation.
+    #
+    # Groups isolate breaking decisions: e.g. if a breakable is transformed into
+    # an actual break, it parent groups might not get broken if the result fits
+    # on the line.
+    #
+    # This is not the case with {nest}: if a the same breakable was in a nest,
+    # the parent group will also be broken.
```

Comment:
This definition doesnt' sit right with me since it doesn't take into account `inconsistent` groups.

The way I see it, `nest`'s main purpose is for indenting small specific part of the output, while `group`'s purpose is to (like mentionned in its name) group elements together allowing us to have parent/child like hierachy.

e.g
```ruby
out.group { # GROUP#1
  out.group { # Child of GROUP#1
  }
  out.nest { # Member of GROUP#1
  }
}
```

Lets hop on a call to discuss this and find a better definition.